src/pages/Teams/__tests__/Teams.test.tsx

```diff
@@ -29,7 +29,21 @@ describe('Teams', () => {
     });
 
     it('should render spinner while loading', async () => {
-        // TODO - Add code for this test
+        jest.spyOn(API, 'getTeams').mockResolvedValue([
+            {
+                id: '1',
+                name: 'Team1',
+            },
+        ]);
+
+        render(<Teams />);
+        
+        expect(screen.getByTestId('spinner')).toBeInTheDocument();
+
+        await waitFor(() => {
+            expect(screen.getByText('Team1')).toBeInTheDocument();
+        });
+        expect(screen.queryByTestId('spinner')).toBeNull();
```

Comment:
after page is loaded, the `spinner` should not exist in the page anymore (`useEffect` working as expected)